ipykernel/zmqshell.py

```diff
@@ -33,7 +33,7 @@
 from IPython.core.magics import MacroToEdit, CodeMagics
 from IPython.core.magic import magics_class, line_magic, Magics
 from IPython.core import payloadpage
-from IPython.core.usage import default_gui_banner
+from IPython.core.usage import default_banner
```

Comment:
I'd prefer that we could do the removal of `default_gui_banner` in the next minor IPython release instead of the next major one (given that's such a small change).

So I propose to enclose this in a try/except block, like this:

```python
try:
    from IPython.core.usage import default_gui_banner as default_banner
except ImportError:
    from IPython.core.usage import default_banner
```